File: TypeScript/src/estimate-purchase.steps.ts
import { then, when, binding } from "cucumber-tsflow";
import { assert } from "chai";

@binding()
export class EstimatePurchaseSteps {
  @when(/^J\'achète (\d*) copie\(s\) \"([^\"]*)\"$/)
  public whenJEcoleDesSorciers(nbCopyOfBook: string, title: string): void {
    throw new Error("Not implemented");
  }

  @then(/^Je dois payer (.*)€$/)
  public thenJeDoisPayer(expectedTotal: string): void {
    throw new Error("Not implemented");
  }
}

File: CSharp/test/Dojo.KataPotter.Test/EstimatePurchaseSteps.cs
namespace Dojo.KataPotter.Test
{
    using TechTalk.SpecFlow;

    [Binding]
    public class EstimatePurchaseSteps
    {
        [When("^J'achète (\\d+) copie\\(s\\) \"(.*)\"$")]
        public void WhenJEcoleDesSorciers(int nbCopyOfBook, string title)
        {
            ScenarioContext.Current.Pending();
        }

        [Then("^Je dois payer (.*)€$")]
        public void ThenJeDoisPayer(decimal expectedTotal)
        {
            ScenarioContext.Current.Pending();
        }
    }
}